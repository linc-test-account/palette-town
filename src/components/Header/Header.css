.header {
  grid-area: "header";
  display: flex;
  flex-direction: row;
  background: hsl(0, 0%, 96%);
  background: var(--header-background);
  overflow: hidden;
  min-height: 50px;
  max-height: 50px;
}

.defaultBrandName {
  font-size: 20px;
  font-weight: bold;
  color: hsl(204, 96%, 42%);
  color: var(--app-brand-color);
  font-family: "Hammersmith One", sans-serif;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0 10px 0 10px;
}

.mobileBrandName {
  font-size: 20px;
  font-weight: bold;
  color: var(--app-brand-color);
  font-family: "Hammersmith One", sans-serif;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0 10px 0 10px;
  display: none;
}

@media screen and (max-width: 700px) {
  .defaultBrandName {
    display: none;
  }
  .mobileBrandName {
    display: flex;
  }
}
