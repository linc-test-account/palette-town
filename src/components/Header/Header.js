import React, { Component } from "react";
import { observer } from "mobx-react";
import HeaderButton from "../HeaderButton/HeaderButton";
import PropTypes from "prop-types";
import styles from "./Header.css";

@observer
class Header extends Component {
  static propTypes = {
    toggleSideNav: PropTypes.func,
    modalHandleClick: PropTypes.func,
    dataStore: PropTypes.object
  };

  render() {
    const { toggleSideNav, modalHandleClick, dataStore } = this.props;
    const heartType = dataStore.palette.favorited === true ? "favorite" : "favorite_border"
    return (
      <div className={styles.container}>
        <HeaderButton
          dataStore={dataStore}
          btnFunction={() => toggleSideNav(true)}
          iconName={"menu"}
        />
        <h1 className={styles.defaultHeading}>Palette Town</h1>
        <h1 className={styles.mobileHeading}>PT</h1>
        <HeaderButton
          dataStore={dataStore}
          btnFunction={() => dataStore.generateNewPalatte()}
          iconName={"palette"}
        />
        <HeaderButton
          dataStore={dataStore}
          btnFunction={() => dataStore.palette.addSwatch()}
          iconName={"add"}
        />
        <HeaderButton
          dataStore={dataStore}
          btnFunction={() => dataStore.palette.reversePalette()}
          iconName={"compare_arrows"}
        />
        <HeaderButton
          dataStore={dataStore}
          btnFunction={() => dataStore.palette.randomizePalette()}
          iconName={"shuffle"}
        />
        <HeaderButton
          dataStore={dataStore}
          btnFunction={() => dataStore.pushToFavorites()}
          iconName={heartType}
          isActive={dataStore.palette.favorited}
        />
        <HeaderButton
          dataStore={dataStore}
          btnFunction={modalHandleClick}
          iconName={"file_download"}
        />
      </div>
    );
  }
}

export default Header;
