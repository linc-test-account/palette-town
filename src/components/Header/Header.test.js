import Header from "./Header";

import React from 'react';
import { shallow } from "enzyme";

describe("Header", () => {
  let props;
  beforeEach(() => {
    props = {
        toggleSideNav: jest.fn(),
        modalHandleClick: jest.fn(),
        dataStore: { palette: {}} 
    };
  });
  it("should render correctly", () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
