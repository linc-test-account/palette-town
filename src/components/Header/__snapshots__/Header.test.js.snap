// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="bars"
  />
  <h1>
    Palette Town
  </h1>
  <h1>
    PT
  </h1>
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="arrow-right"
  />
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="plus"
  />
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="exchange"
  />
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="random"
  />
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="heart-o"
  />
  <HeaderButton
    btnFunction={[Function]}
    dataStore={
      Object {
        "palette": Object {},
      }
    }
    fontAwesomeIcon="download"
  />
</div>
`;
